p4include/psa.p4

```diff
@@ -25,403 +24,518 @@ limitations under the License.
-  ///
-  /// The PSA implements drop as an operation in the PRE. While the
-  /// drop operation can be invoked anywhere in the ingress pipeline,
-  /// the semantics supported by the PSA is that the drop will be at
-  /// the end of the pipeline (ingress or egress).
-  void drop      ();
-
-  /// Clone operation: create a copy of the packet and send it to the
-  ///                  specified port.
-  ///
-  /// The PSA specifies four types of cloning, with the packet sourced
-  /// from different points in the pipeline and sent back to ingress
-  /// or to the buffering queue in the egress (@see CloneMethod_t).
-  ///
-  /// @param clone_method  The type of cloning.
-  /// @param port          The port to send the cloned packet to.
-  ///
-  /// Note: based on discussions with Antonin, we can ignore session now
-  void clone     (in CloneMethod_t clone_method, in PortId_t port);
-
-  /// Clone operation: create a copy of the packet with additional
-  ///                  data and send it to the specified port.
-  ///
-  /// The PSA specifies four types of cloning, with the packet sourced
-  /// from different points in the pipeline and sent back to ingress
-  /// or to the buffering queue in the egress (@see CloneMethod_t).
-  ///
-  /// @param clone_method  The type of cloning.
-  /// @param port          The port to send the cloned packet to.
-  /// @param data additional header data attached to the packet
-  void clone<T>  (in CloneMethod_t clone_method, in PortId_t port, in T data);
-
-
-  /// Resubmit operation: send a packet to the ingress port with
-  ///                     additional data appended
-  ///
-  /// This operation is intended for recursive packet processing.
-  ///
-  /// @param data A header definition that can be added to the set of
-  ///             packet headers.
-  /// @param port The input port at which the packet will be resubmitted.
-  void resubmit<T>(in T data, in PortId_t port);
-
-  /// Recirculate operation: send a post deparse packet to the ingress
-  ///                        port with additional data appended
-  ///
-  /// This operation is intended for recursive packet processing.
-  ///
-  /// @param data A header definition that can be added to the set of
-  ///             packet headers.
-  /// @param port The input port at which the packet will be resubmitted.
-  void recirculate<T>(in T data, in PortId_t port);
-
-  /// Truncate operation
-  ///
-  /// Truncate the outgoing packet to the specified length
-  ///
-  /// @param length packet length
-  void truncate(in bit<32> length);
+extern BufferingQueueingEngine {
+    BufferingQueueingEngine();
+
+  // BufferingQueueingEngine(); /// No constructor. BQE is instantiated
+                                /// by the architecture.
 
-  /*
-  @ControlPlaneAPI
-  {
-  // TBD
-  }
-  */
 }
 
-/* --------------------------------------------------------------------- */
-/*
- * Hashes
- */
+// BEGIN:Clone_extern
+extern clone_out {
```

Comment:
Other externs are capitalized.

Reply:
I am pretty sure the clone, resubmit, and recirculate extern objects will be deleted soon.  @hanw does that match your understanding, too?

Reply:
yes, that's correct.